src/datasets/io/generator.py

```diff
@@ -23,20 +24,22 @@ def __init__(
             keep_in_memory=keep_in_memory,
             streaming=streaming,
             num_proc=num_proc,
+            split=split,
             **kwargs,
         )
         self.builder = Generator(
             cache_dir=cache_dir,
             features=features,
             generator=generator,
             gen_kwargs=gen_kwargs,
+            split=split,
             **kwargs,
         )
 
     def read(self):
         # Build iterable dataset
         if self.streaming:
-            dataset = self.builder.as_streaming_dataset(split="train")
+            dataset = self.builder.as_streaming_dataset(split=self.split)
```

Comment:
Instead of defining and using the `split` parameter as an attribute of the `GeneratorDatasetInputStream` class (and `AbstractDatasetInputStream` and children), we can do differently.

We can define the `split` parameter as an attribute of `GeneratorConfig` and here use it as: `self.builder.config.split`.